clients/src/test/java/org/apache/kafka/common/utils/MockTime.java

```diff
@@ -20,13 +20,20 @@
 public class MockTime implements Time {
 
     private long nanos = 0;
+    private long autoTickMs = 0;
 
     public MockTime() {
         this.nanos = System.nanoTime();
     }
 
+    public MockTime(long autoTickMs) {
+        this.nanos = System.nanoTime();
+        this.autoTickMs = autoTickMs;
```

Comment:
Not sure I see why successive calls to nanoseconds() should result in less time elapsed than successive calls to milliseconds(). Either way, the increase is arbitrary since it has no relation to what the process is actually doing between each call. Perhaps if we want to support nanosecond granularity for auto-tick, the easiest thing to do is just change `autoTickMs` to `autoTickNanos`.